dagql/introspection/types.go

```diff
@@ -749,6 +832,70 @@ func (i *InputValue) DeprecationReason() *string {
 	return &reason.Value.Raw
 }
 
-func (d *Directive) Description() string {
-	return d.description
+func (i *InputValue) Directives() []*DirectiveApplication {
+	return directiveApplications(i.directives)
+}
+
+var _ dagql.Typed = &DirectiveApplication{}
+
+func (d *DirectiveApplication) Type() *ast.Type {
+	return &ast.Type{
+		// Some clients don't like custom introspection types like this :(
+		// NamedType: "__DirectiveApplication",
+		NamedType: "_DirectiveApplication",
+		NonNull:   true,
+	}
+}
```

Comment:
Note to the reviewer: this is very sad :cry: 

Ideally, I'd love to be able to define our own "internal" types, and it looked for a while like I could, but python's graphql-core really did not like this: https://github.com/graphql-python/graphql-core/blob/4e83d4201a2be88832f24c5733c0a1b8568c3708/src/graphql/type/validate.py#L181-L196

So, we can just use a single underscore here, and just modify the codegen for the clients to handle this case.